eng/common/scripts/Package-Properties.ps1

```diff
@@ -129,16 +134,25 @@ function Get-PrPkgProperties([string]$InputDiffJson) {
             if ($shouldInclude) {
                 $packagesWithChanges += $pkg
 
-                if ($pkg.DependentPackages) {
-                    $dependentPackagesForInclusion += $pkg.DependentPackages
+                if ($customDependentPackagesHandler -and $pkg.CanaryPackages) {
+                    $dependentPackagesForInclusion += $pkg.CanaryPackages
                 }
             }
         }
     }
 
-    foreach ($addition in $dependentPackagesForInclusion) {
-        if ($lookup[$addition]) {
-            $packagesWithChanges += $lookup[$addition]
+    # there is a custom function to get dependent packages from a changed package set and diff
+    if ($customDependentPackagesHandler)
+    {
+        $packagesWithChanges += &$DependentPackagesFromPackageSetFn $packagesWithChanges $diff
+    }
+    else {
```

Comment:
Should this be an else? It seems like it should be a union of the two sets of packages.

Reply:
It's intentional, but you could disagree with the intent 👍 

I came at this from the stance of:

> If you don't give me a custom function, I will use the default. Otherwise I will only use what you decide.

Hence the `else` instead of the `or` here. You want a union so that's what I'll go with.

Reply:
Let's start with a union and see how the usages end up looking like to better understand if the custom function needs to override the entire set or not.